src/Domain/Entities/Category.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Domain.Common;
+
+namespace Domain.Entities
+{
+    class Category : CommonEntity
+    {
+        public string Title { get; set; }
+        public string Thumbnail { get; set; }
+        public int? ParentId { get; set; }
+        public virtual ICollection<Category> CategoryChildren { get; set; }
+        public virtual Category CategoryParent { get; set; }
```

Comment:
This will cause self referencing loop. Is there any way we can solve this? @MahdiFaheem @kanismim